### PR TITLE
Correct syntax for `error_bar` of `pygmt.Figure.plot`

### PR DESCRIPTION
**Pull request by yvonnefroehlich**
**Description of proposed changes**

This PR aims to correct the syntax for the `error_bar` parameter of [`pygmt.Figure.plot`](https://www.pygmt.org/dev/api/generated/pygmt.Figure.plot.html#pygmt.Figure.plot) following the upstream GMT documentation https://docs.generic-mapping-tools.org/dev/plot.html#e.

For context see https://github.com/GenericMappingTools/pygmt/issues/2182#issuecomment-1304620327.

**Reminders**

- [ ] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If wrapping a new module, open a 'Wrap new GMT module' issue and submit reasonably-sized PRs.
- [ ] If adding new functionality, add an example to docstrings or tutorials.

**Slash Commands**

You can write slash commands (`/command`) in the first line of a comment to perform
specific operations. Supported slash commands are:

- `/format`: automatically format and lint the code
- `/test-gmt-dev`: run full tests on the latest GMT development version

### CODE REVIEW

File: pygmt/src/plot.py [hunk 1]
```diff
@@ -116,8 +116,8 @@ def plot(self, data=None, x=None, y=None, size=None, direction=None, **kwargs):
         *dx/dy* [Default is no offset]. If *dy* is not given it is set
         equal to *dx*.
     error_bar : bool or str
-        [**+b**\|\ **d**\|\ **D**][**+xl**\|\ **r**\|\ *x0*]\
-        [**+yl**\|\ **r**\|\ *y0*][**+p**\ *pen*].
+        [**x**\|\ **y**\|\ **X**\|\ **Y**][**+a**\|\ **A**]\
+        [**+cl**\|\ **f**][**+n**][**+w**\ *cap*][**+p**\ *pen*].
         Draw symmetrical error bars. Full documentation is at
```
Comment (seisman) [inline pygmt/src/plot.py:121]: **+a** can draw asymmetric error bars, so the description is a little misleading.
```suggestion
        Draw error bars. Full documentation is at
```
We may also need to update the description in the upstream GMT documentation.

Comment (yvonnefroehlich) [reply pygmt/src/plot.py]: > We may also need to update the description in the upstream GMT documentation.

Opened a corresponding PR in the GMT repository https://github.com/GenericMappingTools/gmt/pull/7081.

**Review by seisman — APPROVED** [review]

File: pygmt/src/plot.py [hunk 2]
```diff
@@ -118,9 +118,9 @@ def plot(self, data=None, x=None, y=None, size=None, direction=None, **kwargs):
         *dx/dy* [Default is no offset]. If *dy* is not given it is set
         equal to *dx*.
     error_bar : bool or str
-        [**+b**\|\ **d**\|\ **D**][**+xl**\|\ **r**\|\ *x0*]\
-        [**+yl**\|\ **r**\|\ *y0*][**+p**\ *pen*].
-        Draw symmetrical error bars. Full documentation is at
+        [**x**\|\ **y**\|\ **X**\|\ **Y**][**+a**\|\ **A**]\
+        [**+cl**\|\ **f**][**+n**][**+w**\ *cap*][**+p**\ *pen*].
+        Draw error bars. Full documentation is at
         :gmt-docs:`plot.html#e`.
```
Comment (yvonnefroehlich) [inline pygmt/src/plot.py:124]: @seisman Should I update the syntax for `error_bar` corresponding to the change in https://github.com/GenericMappingTools/gmt/pull/7072? Or should we keep the syntax related to GMT 6.4.0 (for the case PyGMT v0.8.0 will be released before GMT 6.5.0)? 

```suggestion
        [**x**\|\ **y**\|\ **X**\|\ **Y**][**+a**\|\ **A**]\
        [**+cl**\|\ **f**][**+n**][**+w**\ *width*\ [/*cap*]]\
        [**+p**\ *pen*].
        Draw error bars. Full documentation is at
        :gmt-docs:`plot.html#e`.
```

Comment (seisman) [reply pygmt/src/plot.py:124]: I think we should keep the old syntax because GMT 6.5.0 is not released yet.

Comment (yvonnefroehlich) [reply pygmt/src/plot.py:124]: That's fine with me!

**Pull request merged**